pylint/checkers/base/name_checker/checker.py

```diff
@@ -543,7 +549,11 @@ def _should_exempt_from_invalid_name(node: nodes.NodeNG) -> bool:
             warnings = bad_name_group.setdefault(match.lastgroup, [])  # type: ignore[union-attr, arg-type]
             warnings.append((node, node_type, name, confidence))
 
-        if match is None and not _should_exempt_from_invalid_name(node):
+        if (
+            match is None
+            and not disallowed_check_only
```

Comment:
This could've been done in other ways, too, including integrating with `_should_exempt_from_invalid_name` above but I chose this for simplicity. Lmk if another implementation makes more sense.